src/esn.py

```diff
@@ -132,19 +179,181 @@ def _forward(self, ut: Array, x_init: Optional[Array] = None,
                 y = np.dot(self.w_out, np.concatenate([x, u], axis=0))
             else:
                 y = np.dot(self.w_out, x)
-            # collect state (if desired) and output
+            yt.append(y)     
             if collect_states:
                 xt.append(x)
-            yt.append(y)
         # collect outputs and reservoir states into matrices
         yt = np.concatenate(yt, axis=1).T
+        
         if collect_states:
             xt = np.concatenate(xt, axis=1).T
             return xt, yt
         else:
             return yt
 
-    def harvest_states(self, ut: Array, x_init: Optional[Callable] = None,
+    def _forwardRFC(self, ut: Array, x_init: Optional[Array] = None, 
```

Comment:
perhaps it's better to have the type of conceptor be a parameter in the trainingConfiguration (or maybe even esnConfiguration). I would think that we usually know what kind of conceptors we want to use before setting up the ESN in the first place - and then the type of conceptor won't change anymore - so we could just put it into the esnConfig, what do you think?

Reply:
because then we can simply call the normal functions (like `forward` or `harvest_states`) while the internal state of the ESN instance will handle the details